STMReco/src/ZPAlg.cc

```diff
@@ -0,0 +1,83 @@
+///////////////////////////////////////
+//
+// Zero-suppression algorithm for STM
+// Original authors: Claudia Alvarez-Garcia, Alex Kesharvarzi, and Mark Lancaster
+// Adapated for Offline: Andy Edmonds
+//
+#include "Offline/STMReco/inc/ZPAlg.hh"
+
+namespace mu2e {
+
+  int16_t* ZPAlg::ZeroSuppress(const int16_t* ADC, unsigned long int n, std::vector<size_t>& starts, std::vector<size_t>& ends){
+    //n number of elements to apply zero sup
+
+    //----------------------- PEAKS SUPPRESSION ALGORITHM-----------------------------------------
+
+    //Calculate the gradient for ADC values separated a distance window (in time would be window * tadc)
+    int window=100;
```

Comment:
is the window size intended to be constant?  If so, consider making the ADC data an array.  If it might change, consider making it a vector.   If it must be passed by pointer, please pass the length with it.  These changes would avoid assuming the array length (which is unconstrained by the interface).

Reply:
I'm not sure to be honest. This code is likely to move to online so we will keep this in mind

Reply:
This code returns a pointer to something.  You should annotate the code to make it clear whether or not the caller receives ownership of memory ( ie is it their job to delete anything?).